rescript/_utilities.py

```diff
@@ -21,15 +22,29 @@
 }
 
 
-def _find_lca_in_series(t1, t2):
-    '''
-    Find least common ancestor between two semicolon-delimited strings.
-    Input consists of two series containing index "Taxon".
-    Returns the LCA of those taxonomies as an iterator.
-    '''
-    t1 = t1['Taxon']
-    t2 = t2['Taxon']
-    return _find_lca([t1, t2])
+# modified version of _find_lca that prioritizes majority and supersets
+def _find_super_lca(taxa, collapse_substrings=True):
```

Comment:
I'm glad you added this! I had been meaning to inquire about such an approach, as sintax/utax (if I remember correctly) does something similar. That is, going for a more specific taxonomy string when it does not contradict upper-level taxonomy.